### PR TITLE
feat(UKTV): add presence

### LINKED ISSUE #6514: UKTV
### Website name

UKTV Play

### Website URL

https://uktvplay.uktv.co.uk/

### Website logo

https://upload.wikimedia.org/wikipedia/en/2/2b/UKTV_logo.png

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [X] It is region restricted

### Description

Streaming service - a "now watching" with elapsed time / time left, a browsing etc. Similar to Netflix

### PR DESCRIPTION
**Pull request by DarkVillager**
## Description 

Resolves #6514 
Added presence, with privacy, buttons and cover settings.

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/U/UKTV/presence.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+const presence = new Presence({
+		clientId: "1002899869599551508",
+	}),
+	strings = presence.getStrings({
+		play: "general.playing",
+		pause: "general.paused",
+		browsing: "general.browsing",
+	});
```
Comment (EncryptedDev) [inline websites/U/UKTV/presence.ts:8]: If you have strings, you should include a multiLang setting and use the Awaited type for strings

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/U/UKTV/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		"id": "638080361179512853"
+	},
+	"service": "UKTV",
+	"description": {
+		"en": "UKTV Play, the free on demand TV service. Watch the best comedy, drama, entertainment and documentaries from Dave, Drama, W and Yesterday. Watch on your TV, mobile, tablet, or online."
+	},
+	"url": "uktvplay.uktv.co.uk",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/wjqmEQY.png",
+	"thumbnail": "https://i.imgur.com/wjqmEQY.png",
```
[the remark this answers is not in the document]
Comment (DarkVillager) [reply websites/U/UKTV/dist/metadata.json:14]: > The thumbnail is the same as the logo? 🤔

oops

File: websites/U/UKTV/presence.ts (hunk 1 above)
Comment (DarkVillager) [reply websites/U/UKTV/presence.ts:8]: > If you have strings, you should include a multiLang setting and use the Awaited type for strings

but no 1000 users?

Comment (EncryptedDev) [reply websites/U/UKTV/presence.ts:8]: That rule is for adding custom strings for said presence, not adding multiLang in general.

Comment (DarkVillager) [reply websites/U/UKTV/presence.ts:8]: > That rule is for adding custom strings for said presence, not adding multiLang in general.

ok thx for clearing that up.

File: websites/U/UKTV/presence.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
+const presence = new Presence({
+	clientId: "1002899869599551508",
+});
+
+async function getStrings() {
+	return presence.getStrings(
+		{
+			play: "presence.playback.playing",
+			pause: "presence.playback.paused",
+			browse: "presence.activity.browsing",
```
Comment (EncryptedDev) [inline websites/U/UKTV/presence.ts:10]: `presence.` strings will be removed soon, so using `general.` is btter
```suggestion
			play: "general.playing",
			pause: "general.paused",
			browse: "general.browsing",
```

File: websites/U/UKTV/presence.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		presenceData.smallImageKey = "search";
+	} else if (video?.duration) {
+		const titles = document.querySelector(
+			'[class="vod-episode__title margin-bottom--xxs invert-text semi-bold"]'
+		)?.textContent;
+		presenceData.details = titles;
+
+		presenceData.smallImageKey = video.paused ? "pause" : "play";
+		presenceData.smallImageText = video.paused
+			? (await strings).pause
```
Comment (EncryptedDev) [inline websites/U/UKTV/presence.ts:51]: With the Awaited type on L16, you do not need to await localised strings

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Other than this, LGTM

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**